src/Notifier.js

```diff
@@ -133,29 +155,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
+                // Attempt to get permission from user
+                var self = this;
+                global.Notification.requestPermission(function(result) {
+                    if (result === 'denied') {
                         dis.dispatch({
                             action: "notifier_enabled",
-                            value: true
+                            value: false
                         });
+                        self.setToolbarHidden(true, false);
```

Comment:
I'm running out of ways to say this:  Leave it alone for now, and make it a separate PR.